### PR TITLE
CA-309809: avoid a stuck GFS2 mount by checking quorate state before PBD plug

### PR DESCRIPTION
**Pull request by edwintorok**
Also update unit test to return a dummy quorate diagnostics result.

### CODE REVIEW

**Comment by MarkSymsCtx** [comment]
I think I have to defer to @gaborigloi on this review, it's beyond my ability to parse OCaml.

File: ocaml/xapi/xapi_clustering.ml [hunk 1]
```diff
@@ -241,6 +221,46 @@ let rpc ~__context =
...
+  begin match get_required_cluster_stacks ~__context ~sr_sm_type with
+    | [] -> ()
+    | required_cluster_stacks ->
+      (* One of these [required_cluster_stacks] should be configured and running *)
+      let cluster_stack_of ~cluster_host =
+        let cluster = Db.Cluster_host.get_cluster ~__context ~self:cluster_host in
+        Db.Cluster.get_cluster_stack ~__context ~self:cluster
+      in
+      let error_no_cluster_host_found condition =
+        debug "No_cluster_host found%s" condition;
```
Comment (lindig) [inline ocaml/xapi/xapi_clustering.ml]: Missing a space in the format string?

File: ocaml/xapi/xapi_clustering.ml [hunk 2]
```diff
@@ -241,6 +221,46 @@ let rpc ~__context =
...
+        raise Api_errors.(Server_error (no_compatible_cluster_host, [Ref.string_of host]))
+      in
+      match find_cluster_host ~__context ~host with
+        | Some cluster_host when (List.mem (cluster_stack_of ~cluster_host) required_cluster_stacks) ->
```
Comment (lindig) [inline ocaml/xapi/xapi_clustering.ml]: The indentation for `match` looks different from above. Maybe run this through xcp-indent

Comment (edwintorok) [reply ocaml/xapi/xapi_clustering.ml]: Done, I think this code got moved around quite a bit.

**Pull request merged**